### DISCUSSION TITLE
Bug update time SONOFF SNZB-02D

### DISCUSSION THREAD

**Discussion by Kap3** [Koenkk/zigbee2mqtt#21352] [Devices]
Hi, I use zigbee2mqtt for connect my temperature sensor SONOFF SNZB-02D, as concetrator I use ZBDongle-E with RCP firmware.
I have a problem with update time, the device continuously updates the data ( check the log) so It finish the battery in 2 weeks.
I use last version of zigbee2mqtt ( 1.35.3-1 ), in this version I found the fix:

https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.35.3
https://github.com/Koenkk/zigbee-herdsman-converters/issues/7013 Improve SONOFF SNZB-02 reporting intervals (@enryIT)

For sonoff WITHOUT "D" ( where D is display version).
Is there the same bug in my device?
Can you solve it?
Thanks

LOG:

> Received 10:29:52 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20.2

Received 10:32:54 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20.2

Received 10:33:24 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 10:34:53 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 10:43:27 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 10:44:56 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 10:48:28 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 10:53:29 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 10:54:59 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

Received 11:00:00 AM

    QoS: 0
    Payload:

    battery: 80
    humidity: 53.3
    linkquality: 255
    temperature: 20

**Comment by Kap3** [comment]
Another log:

info 2024-02-11 11:55:17MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":53.3,"linkquality":255,"temperature":20}'
info 2024-02-11 11:58:49MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":53.3,"linkquality":255,"temperature":20}'
info 2024-02-11 11:59:18MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":20}'
info 2024-02-11 12:03:50MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":20}'
info 2024-02-11 12:04:20MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":20}'
info 2024-02-11 12:08:51MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":20}'
info 2024-02-11 12:09:21MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":20}'
info 2024-02-11 12:10:53MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":20}'
info 2024-02-11 12:13:33MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:14:22MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:18:35MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:19:24MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:23:41MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:24:25MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:38:40MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:43:41MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:44:30MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:48:43MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
info 2024-02-11 12:49:32MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'

**Comment by enryIT** [comment]
Hello,

I made that commit, both  SNZB-02 and SNZB-02D share the same battery type CR2450, the only difference is the display in the "D" version and a much more recent temperature and humidity sensor, still for the "D" one.
The reporting intervals before were inusable, I just made them more properly useful.

Why are you saying that your batteries will last 2 weeks?
I tested on 2 different SNZB-02 models the publication changes and I measured a 15% mean increase in battery consumption over a period of 2 months, meaning that I lost 6% in 2 months with previous values and 7% in 2 month after the changes.

Now, since temperature and humidity are totaly separeted zigbee clusters, their timers are not shared, this means that you can receive an update for temperature and immediately after an update for humidity or battery percentage or voltage or even link quality.
Anyway you're logs are weird, you should only receive an update if temperature changes more than .2°C or humidity more than 1% or every 5 minutes regardless of deltas, your logs says that you are receiving updates without changes in any of the available attributes, for example:

> info 2024-02-11 12:24:25MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
> info 2024-02-11 12:38:40MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
> info 2024-02-11 12:43:41MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
> info 2024-02-11 12:44:30MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
> info 2024-02-11 12:48:43MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'
> info 2024-02-11 12:49:32MQTT publish: topic 'zigbee2mqtt/Temp 1', payload '{"battery":80,"humidity":54.3,"linkquality":255,"temperature":19.8}'

The first update (12:24:25 - 12:38:40) is missing something in between, more than 14 minutes without updates?
The second update is correct (12:38:40 - 12:43:41) is correct, 5 minutes have passed without changes, update is sent but we don't know for which zigbee cluster (temp or humidity)
The third one (12:43:41 - 12:44:30) could be correct, it depends on which cluster was updated on the previous one.
The fourth one (12:44:30 - 12:48:43) isn't correct, temperature nor humidity changed, 5 minutes have not passed.

This is a correct example of my instance:
Temperature:

> debug 2024-02-11 14:51:58: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msTemperatureMeasurement', data '{"measuredValue":1962}' from endpoint 1 with groupID 0
> debug 2024-02-11 14:56:59: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msTemperatureMeasurement', data '{"measuredValue":1962}' from endpoint 1 with groupID 0
> debug 2024-02-11 15:01:59: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msTemperatureMeasurement', data '{"measuredValue":1959}' from endpoint 1 with groupID 0
> debug 2024-02-11 15:06:59: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msTemperatureMeasurement', data '{"measuredValue":1959}' from endpoint 1 with groupID 0
> debug 2024-02-11 15:11:59: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msTemperatureMeasurement', data '{"measuredValue":1959}' from endpoint 1 with groupID 0

Humidity:

> debug 2024-02-11 14:52:29: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msRelativeHumidity', data '{"measuredValue":6231}' from endpoint 1 with groupID 0
> debug 2024-02-11 14:57:29: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msRelativeHumidity', data '{"measuredValue":6242}' from endpoint 1 with groupID 0
> debug 2024-02-11 15:02:29: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msRelativeHumidity', data '{"measuredValue":6231}' from endpoint 1 with groupID 0
> debug 2024-02-11 15:07:29: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msRelativeHumidity', data '{"measuredValue":6242}' from endpoint 1 with groupID 0
> debug 2024-02-11 15:12:29: Received Zigbee message from 'Termoigrometro Studio', type 'attributeReport', cluster 'msRelativeHumidity', data '{"measuredValue":6231}' from endpoint 1 with groupID 0

Could you please post a screenshot of your sensor 's reporting settings?
This is what it sould look like.
![Screenshot 2024-02-11 182118](https://github.com/Koenkk/zigbee2mqtt/assets/590989/de9fa074-4b7d-41f4-b2e8-4b90728f3afa)

**Reply by Kap3** [reply]
Ciao Enrico,
Maybe 2 weeks is pessimistic. I change battery 10 February. And this is third battery from October...
This is my settings 
Thanks 
![Screenshot_20240212_092157_Home Assistant](https://github.com/Koenkk/zigbee2mqtt/assets/32192829/5e42164c-b21c-421a-9d1c-c5b1cc0bcd0e)

**Reply by enryIT** [reply]
Ciao,

the reporting intervals match the correct ones, have you tried repairing the sensor?
Also, the changes are live from 1 week ago, if you changed already 2 batteries in 4 months it means thtat the problem relies somewhere else. 

My SNZB-02 will consume a battery in a year and a half.

**Reply by Kap3** [reply]
What do you mean "repair the sensor"?
the problem has been there since I bought the sensor (September?) if I disconnect the sensor from home assistant the battery works fine.

I have one moes temperature sensor with a little cr2032 and it works fine!

**Reply by enryIT** [reply]
I mean delete it from zigbee2mqtt and pair it again, I think something has messed up in zigbee's cluster configurations since it's updating with weird timings that don't match with what your reporting interval configuration is.

Indeed, if the problem you're having is happening from more than a week ago it's not because of the patch you mentioned but something else. I understand that this battery issue is present from the day you associated this sensor with z2m.

**Reply by Kap3** [reply]
I solved it.
I deleted all parameter and left only the first, then I push yellow button RECONFIG. 
Now It works fine. I test the battery ​I'll test the battery in the next weeks.
![immagine](https://github.com/Koenkk/zigbee2mqtt/assets/32192829/637e689f-ebf8-47da-822d-173966226f0a)

![immagine](https://github.com/Koenkk/zigbee2mqtt/assets/32192829/95682b0c-05cb-44a2-9baf-53d40e487106)

Thanks
